thewalrus/symplectic.py

```diff
@@ -321,3 +321,35 @@ def is_symplectic(S, rtol=1e-05, atol=1e-08):
     Omega = sympmat(nmodes)
 
     return np.allclose(S.T @ Omega @ S, Omega, rtol=rtol, atol=atol)
+
+def autonne(A, rtol=1e-05, atol=1e-08, svd_order=True):
+    r"""Autonne-Takagi decomposition of a complex symmetric (not Hermitian!) matrix.
+
+    Args:
+        A (array): square, symmetric matrix
+        rtol (float): the relative tolerance parameter between ``A`` and ``A.T``
+        atol (float): the absolute tolerance parameter between ``A`` and ``A.T``
+        svd_order (boolean): whether to return result by ordering the singular values of ``A`` in descending (``True``) or asceding (``False``) order.
+
+    Returns:
+        tuple[array, array]: (r, U), where r are the singular values,
+            and U is the Autonne-Takagi unitary, such that :math:`A = U \diag(r) U^T`.
```

Comment:
What does the `\diag(r)` in  `:math:'A = U \diag(r) U^T'` print? I don't think that's a regular latex command, or does it somehow work differently in Sphinx?

Reply:
Nothing really. Typically we just use a macro defined somewhere to replace \diag to \text{diag}.